src/backlinks/handler.js

```diff
@@ -139,13 +139,50 @@ export default async function auditBrokenBacklinks(message, context) {
       auditResult,
     };
 
-    await dataAccess.addAudit(auditData);
+    const audit = await dataAccess.addAudit(auditData);
     const data = {
       type,
       url: site.getBaseURL(),
       auditContext,
       auditResult,
     };
+
+    const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(siteId, 'NEW');
+    let brokenBacklinksOppty = opportunities.find((oppty) => oppty.getType() === 'broken-backlinks');
+
+    if (!brokenBacklinksOppty) {
+      const opportunityData = {
+        siteId: site.getId(),
+        auditId: audit.getId(),
```

Comment:
If the "brokenBacklinksOppty" already exists, I think, we would need to update the auditId in that case also.

Reply:
good catch! Yes absolutely